contrib/zpages/src/main/java/io/opencensus/zpages/TracezHttpHandler.java

```diff
@@ -0,0 +1,111 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An {@code HttpHandler} that displays information about active and sampled spans recorded using
+ * the OpenCensus library.
+ *
+ * <p>Example usage with automatic context propagation:
+ *
+ * <pre>{@code
+ * public class Main {
+ *   public static void main(String[] args) throws Exception {
+ *     HttpServer server = HttpServer.create(new InetSocketAddress(8000), 10);
+ *     TracezHttpHandler.register(server);
+ *     server.start();
+ *   }
+ * }
+ * }</pre>
+ */
+public final class TracezHttpHandler implements HttpHandler {
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final String TRACEZ_URL = "/tracez";
+  private static final String HTTP_SERVER_SPAN_NAME = "HttpServer/tracez";
+  private final TracezPageFormatter pageFormatter;
+
+  /** Constructs a new {@code TracezHttpHandler}. */
+  private TracezHttpHandler() {
+    ExportComponent exportComponent = Tracing.getExportComponent();
+    this.pageFormatter =
+        TracezPageFormatter.create(
+            exportComponent.getRunningSpanStore(), exportComponent.getSampledSpanStore());
+    Tracing.getExportComponent()
+        .getSampledSpanStore()
+        .registerSpanNamesForCollection(Arrays.asList(HTTP_SERVER_SPAN_NAME));
+  }
+
+  /**
+   * Registers the tracez {@code HttpHandler} to the given {@code HttpServer}.
+   *
+   * @param server the server that exports the tracez page.
+   */
+  public static void register(HttpServer server) {
+    server.createContext(TRACEZ_URL, new TracezHttpHandler());
+  }
+
+  @Override
+  public final void handle(HttpExchange httpExchange) throws IOException {
+    try (Scope ss =
+        tracer
+            .spanBuilderWithExplicitParent(HTTP_SERVER_SPAN_NAME, null)
+            .setRecordEvents(true)
+            .startScopedSpan()) {
+      tracer
+          .getCurrentSpan()
+          .addAttributes(
+              ImmutableMap.<String, AttributeValue>builder()
+                  .put(
+                      "RequestMethod",
+                      AttributeValue.stringAttributeValue(httpExchange.getRequestMethod()))
+                  .build());
+      httpExchange.sendResponseHeaders(200, 0);
+      pageFormatter.emitHtml(queryToMap(httpExchange), httpExchange.getResponseBody());
+      httpExchange.close();
+    }
+  }
+
+  private static Map<String, String> queryToMap(HttpExchange httpExchange) {
+    String query = httpExchange.getRequestURI().getQuery();
+    if (query == null) {
+      return Collections.emptyMap();
+    }
+    Map<String, String> result = new HashMap<String, String>();
+    for (String param : query.split("&")) {
+      String[] pair = param.split("=");
+      if (pair.length > 1) {
+        result.put(pair[0], pair[1]);
+      } else {
+        result.put(pair[0], "");
+      }
+    }
+    tracer.getCurrentSpan().addAnnotation("Finish to parse query components.");
```

Comment:
Was debugging. I will leave some instrumentation because it is nice to have it instrumented :)